### PR TITLE
Add 0.4.12 release notes

### PR DESCRIPTION
**Pull request by sofroniewn**
# Description
Adds 0.4.12 release notes. Maybe @alisterburt or @jni can take a pass through the PR classification and try writing a highlights section - I think this is mainly a small bug fix release, though if there are highlights / API changes I'm missing can someone please call them out. Thanks!!

### CODE REVIEW

**Comment by jni** [comment]
@sofroniewn I've pushed some minor changes.

**Review by jni — APPROVED** [review]

**Comment by psobolewskiPhD** [comment]
Wait, my heart is a flutter 😍 
...but did I really review something?
It must have been on accident!
```
- [psobolewskiPhD](https://github.com/napari/napari/commits?author=psobolewskiPhD) - @psobolewskiPhD
```

**Comment by sofroniewn** [comment]
> ...but did I really review something?
> It must have been on accident!

@psobolewskiPhD it's all automated so you must have! Thanks for your review contribution!! ❤️  Here's to many more 🎉 🚀

**Comment by sofroniewn** [comment]
Thanks @jni for update, i'll merge this once test pass!!

**Comment by psobolewskiPhD** [comment]
> > ...but did I really review something?
> > It must have been on accident!
> 
> @psobolewskiPhD it's all automated so you must have! Thanks for your review contribution!! ❤️ Here's to many more 🎉 🚀

That can't be right though?! Maybe someone misclicked something?
🤯

**Comment by sofroniewn** [comment]
> That can't be right though?! Maybe someone misclicked something?

I think it counts any comments on PRs as reviews

**Comment by psobolewskiPhD** [comment]
🦑

File: docs/release/release_0_4_12.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+- Fix problem with local function signal binding (#3352)
+- Fix __getattr__ in WorkerBase (#3368)
+- Fix off-by-one bug in extent of Image and Labels layers (#3381)
+- Try to fix bundle building. (#3403)
+- Fix to_labels default output shape (#3412)
+- Fix lazy load console (#3419)
+- Fix teardown of menus to prevent widget test leaks (#3433)
+- Fix off-by one error in Dims.range for non pixel-based layers (#3444)
+- Fix naming inconsistency for windows bundle (#3476)
+
```
Comment (alisterburt) [inline docs/release/release_0_4_12.md:72]:
```suggestion
- Add ability to provide empty data to vectors layer (#2995)
```

File: docs/release/release_0_4_12.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+See the full list of merged pull requests below for further delails!
+
+## New Features
+- Add progress bar when opening list of files (#3355)
+- Add right-click context menu to convert label data type. (#3402)
+
+
+## Improvements
+- Support for `Future` return type in magicgui widget  (#2581)
+- Add ability to provide empty data to vectors layer (#2995)
```
Comment (alisterburt) [inline docs/release/release_0_4_12.md:28]:
```suggestion
```

Comment (alisterburt) [reply docs/release/release_0_4_12.md:28]: move to bugfix?

File: docs/release/release_0_4_12.md [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+- Don't register dask cache globally - but do use cache as context manager when slicing (#3285)
+- Hide or Destroy dock widgets (#3331)
+- `_vispy` module cleanup (#3333)
+- Update ubuntu image to 18.04 (#3348)
```
Comment (alisterburt) [inline docs/release/release_0_4_12.md:32]:
```suggestion
```

move to build tools/support?

File: docs/release/release_0_4_12.md [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+- Move 8s tiemout to github actions only (#3457)
+- Move icons to package src (#3462)
+- Fix image display for plugin installation page. (#3480)
+- Fix broken links in documentation (#3481)
+- Update formatting in README.md (#3482)
+- Fix formatting of Layers docstrings (#3483)
+- Add links to napari repo in README (#3484)
+- Bump minimum NumPy requirement to 1.18 (as per NEP29) (#3485)
+- Remove make_napari_viewer in vispy tests. (#3486)
+
```
Comment (alisterburt) [inline docs/release/release_0_4_12.md:109]:
```suggestion
- Update ubuntu image to 18.04 (#3348)
```

File: docs/release/release_0_4_12.md [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+For more information, examples, and documentation, please visit our website:
+https://github.com/napari/napari
+
+## Highlights
+
+This is a bug fix release with many minor improvements and bug fixes. The user
+experience for users of dask arrays might be significantly improved by a new
+approach to setting the contrast limits based on the current slice (#3425).
+
```
Comment (alisterburt) [inline docs/release/release_0_4_12.md:18]:
```suggestion
A progress bar will now display when opening multiple files (#3355). Thanks to first-time contributor @tibuch the data type of labels layers can now be converted from a context menu on the layer list (#3402).
```

**Review by alisterburt — COMMENTED** [review]
quick pass with some minor changes, thanks @sofroniewn and @jni !!

**Comment by sofroniewn** [comment]
@alisterburt i've unfortunately already merged this, can you make a PR!

**Comment by alisterburt** [comment]
actual photo of me 
image

will do!

**Pull request merged**